server/plugin.go

```diff
@@ -599,22 +670,84 @@ func (p *Plugin) HasGroupHook(ctx context.Context, user *gitlab.UserInfo, namesp
 	return found, err
 }
 
-func (p *Plugin) checkAndRefreshToken(token *oauth2.Token) (*oauth2.Token, error) {
-	// If there is only one minute left for the token to expire, we are refreshing the token.
-	// The detailed reason for this can be found here: https://github.com/golang/oauth2/issues/84#issuecomment-831492464
-	// We don't want the token to expire between the time when we decide that the old token is valid
-	// and the time at which we create the request. We are handling that by not letting the token expire.
-	if time.Until(token.Expiry) <= 1*time.Minute {
-		conf := p.getOAuthConfig()
-		src := conf.TokenSource(context.Background(), token)
-		newToken, err := src.Token() // this actually goes and renews the tokens
-		if err != nil {
-			return nil, errors.Wrap(err, "unable to get the new refreshed token")
+func (p *Plugin) refreshToken(userInfo *gitlab.UserInfo, token *oauth2.Token) (*oauth2.Token, error) {
+	conf := p.getOAuthConfig()
+	src := conf.TokenSource(context.Background(), token)
+
+	newToken, err := src.Token() // this actually goes and renews the tokens
+
+	if err != nil {
+		if strings.Contains(err.Error(), "\"error\":\"invalid_grant\"") {
+			p.handleRevokedToken(userInfo)
 		}
-		if newToken.AccessToken != token.AccessToken {
-			return newToken, nil
+		return nil, errors.Wrap(err, "unable to get the new refreshed token")
+	}
+
+	if newToken.AccessToken != token.AccessToken {
+		p.API.LogDebug("Gitlab token refreshed.", "UserID", userInfo.UserID, "Gitlab Username", userInfo.GitlabUsername)
```

Comment:
We want to avoid logging user-identifying information like usernames to the logs

```suggestion
		p.API.LogDebug("Gitlab token refreshed.", "UserID", userInfo.UserID)
```

Also I think this may be more spammy than useful since this will be logged per user every hour. @hanzei What are your thoughts on having this log message here?

Reply:
Also, should we also be comparing `RefreshToken` to check if we need to store the token? Not sure if it's possible for that to change but not the access token

Reply:
This is all left over from the previous token refresh code before I started working on the issue. The `RefreshToken` should not change unless the `AccessToken` changes, though in real use we always receive a different token so the entire check could be unnecessary.

Reply:
Committed your suggested change to remove the GitLab username.